src/groovy/com/unifina/api/UpdateProductCommand.groovy

```diff
@@ -47,10 +47,7 @@ class UpdateProductCommand {
 		product.previewStream = previewStream
 		product.previewConfigJson = previewConfigJson
 
-		if (product.pricePerSecond > 0 && product.state == Product.State.NOT_DEPLOYED) {
-			if (pricePerSecond == 0) {
-				throw new InvalidStateException("Cannot update paid Product (price > 0) to be free (price = 0)")
-			}
+		if (product.state == Product.State.NOT_DEPLOYED) {
```

Comment:
Have the implications of this change been taken into consideration? There might have been a reason for not allowing paid products to be made free afterwards.

Reply:
@harbu Not 100% sure what the original reason for this limitation were, but probably this:

Not having to deal with this particular corner case was a handy shortcut back when we were trying to ship the Marketplace in a hurry.

This is not a valid reason anymore, and I can’t think of any technical reason either right now.

Reply:
You could run into problems with existing subscribers. If I remember correctly, the way in which paid subscriptions and permissions are represented in the data model differs from the way in which free subscriptions / permissions are represented.

Reply:
Can you point out how exactly a problem would occur?

Reply:
What should we do with the contract when paid product is updated to free? Do we just let it hang there or is it removed from the chain? There's also validation logic for pricePerSecond in the contract: https://github.com/streamr-dev/marketplace-contracts/blob/master/contracts/Marketplace.sol#L232

Reply:
I searched through Technical Debt tickets, closest thing I could find related to this is CORE-1322, which is worth a read.

 A couple of things spring to mind:
- The streams of free products have associated anonymous READ permissions in the system that are not necessarily present for paid products. These get created when the free product is deployed / published. (And revoked when the free product is undeployed.)
- Subscriptions of paid products are of class `PaidSubscription` and those of free products are `FreeSubscription`. You could potentially run into problems if a `PaidSubscription` is suddenly referring to a free Product.

It's a long time since I touched this code, so I can't really come up with anything more concrete. All I know is that there are notable differences in how free and paid products are handled and the associated domain objects that are created.

I would say it's okay to proceed with this but expect to discover some bugs later on. At this point in time, without further knowledge, it's probably the practical way forward.

Reply:
Proceeding with this to unblock frontend development. As Eric noted, this may lead to some problems in free vs. paid products behaving differently, but let's resolve any issues if/when we hit them.